src/Providers/MoonGuardServiceProvider.php

```diff
@@ -16,13 +22,21 @@ public function boot(): void
         $this->publishCommands();
         $this->publishRoutes();
         $this->publishViews();
+        $this->publishAssets();
     }
 
     public function register(): void
     {
         $this->mergeConfigFrom(__DIR__ . '/../../config/moonguard.php', 'moonguard');
 
         $this->app->register(EventServiceProvider::class);
+
+        $this->app->bind(MoonGuardSite::class, config('moonguard.site.model'));
+        $this->app->bind(MoonGuardUser::class, config('moonguard.user.model'));
+        $this->app->bind(MoonGuardUptimeCheck::class, config('moonguard.uptime_check.model'));
+        $this->app->bind(MoonGuardSslCertificateCheck::class, config('moonguard.ssl_certificate_check.model'));
+        $this->app->bind(MoonGuardExceptionLog::class, config('moonguard.exceptions.exception_log.model'));
+        $this->app->bind(MoonGuardExceptionLogGroup::class, config('moonguard.exceptions.exception_log_group.model'));
```

Comment:
Why is it necessary to bind each Model with the Configuration Model?

Reply:
Did you have any error by not doing this? is this related to the bug we saw on friday meeting?

Reply:
yeah, as we are passing the interface on the events, Laravel will try to inject the dependency. So, we need to bind the model to the interface so Laravel knows what class should instantiate when we use the interface.